src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -14566,17 +14572,19 @@ void VRFCondenserEquipment::VRFOU_CalcCompH(
         CompEvaporatingCAPSpd(CounterCompSpdTemp) =
             this->CoffEvapCap * this->RatedEvapCapacity * CurveValue(state, this->OUCoolingCAPFT(CounterCompSpdTemp), T_discharge, T_suction);
 
-        if ((Q_evap_req * C_cap_operation) <= CompEvaporatingCAPSpd(CounterCompSpdTemp)) {
+        if ((Q_evap_req * C_cap_operation) <= CompEvaporatingCAPSpd(CounterCompSpdTemp) + CompEvaporatingPWRSpd(CounterCompSpdTemp)) {
             // Compressor Capacity is greater than the required, finish Iteration DoName2
 
             if (CounterCompSpdTemp > 1) {
                 // Compressor runs at higher speed than min speed
                 CompSpdLB = CounterCompSpdTemp - 1;
                 CompSpdUB = CounterCompSpdTemp;
 
-                CompSpdActual = this->CompressorSpeed(CompSpdLB) + (this->CompressorSpeed(CompSpdUB) - this->CompressorSpeed(CompSpdLB)) /
-                                                                       (CompEvaporatingCAPSpd(CompSpdUB) - CompEvaporatingCAPSpd(CompSpdLB)) *
-                                                                       (Q_evap_req * C_cap_operation - CompEvaporatingCAPSpd(CompSpdLB));
+                CompSpdActual = this->CompressorSpeed(CompSpdLB) +
+                                (this->CompressorSpeed(CompSpdUB) - this->CompressorSpeed(CompSpdLB)) /
+                                    (CompEvaporatingCAPSpd(CompSpdUB) + CompEvaporatingPWRSpd(CompSpdUB) -
+                                     (CompEvaporatingCAPSpd(CompSpdLB) + CompEvaporatingPWRSpd(CompSpdLB))) *
+                                    (Q_evap_req * C_cap_operation - (CompEvaporatingCAPSpd(CompSpdLB) + CompEvaporatingPWRSpd(CompSpdLB)));
```

Comment:
Not sure what this math represents. It looks like:

CompSpdActual = CompSpdLow + ((CompSpdHi - CompSpdLo) / {some sum of capacity + power)) * something

Do I have that right? That seems wierd for a compressor speed calculation but I'll leave that to you.

Reply:
it intends to linearly interpolate between the CompSpdLo and CompSpdHi. 
Previously it tries to match the demand of `TU_load + Pipe_Q - Ncomp `to a compressor output of `CompEvaporatingCAPSpd(i)` where i is the correct speed level. There's some linear interpolation going on when the demand does not equal to the compressor output at discrete speed level. 
Now it tries to match `TU_load + Pipe_Q` to `CompEvaporatingCAPSpd(i) + CompEvaporatingPWRSpd(i)`. Ncomp term is the compressor power. CompEvaporatingPWRSpd(i) is the compressor power curve.

The two approaches are kinda equivalent but the second approach matches the quantity used in the limitTUCapacity calculation, so I can reason about where the related supplemental coil calculation went wrong.

I might change it back to the old way after I finish all the modifications.

Reply:
reverted it back to `Q_evap_req = TU_load + Pipe_Q`, also this change above to calculate `CompSpdActual`